src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,192 @@
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Returns an iterator over elements of type {@code duke.task.Task}.
+     *
+     * @return an Iterator.
+     */
+    @Override
+    public Iterator<Task> iterator() {
+        return tasks.iterator();
+    }
+
+    /**
+     * Add a ToDo to List of Task.
+     * @param description String name of task.
+     * @param isDone Status of the task (Done or Undone).
+     * @return Task that was added successfully.
+     */
+    public Task addToDo(String description, boolean isDone) {
+        ToDo toDo = new ToDo(description, isDone);
+        this.addTask(toDo);
+        return toDo;
+    }
+
+    /**
+     * Add a ToDo to List of Task.
+     * @param parameters String that contain all necessary fields to create a ToDo for adding.
+     * @return Task that was added successfully.
+     * @throws ChatBotParameterException when parameters String lack information to create a ToDo.
+     */
+    public Task addToDo(String parameters) throws ChatBotParameterException {
+        String[] parametersArr = Parser.parseToDo(parameters);
+        return this.addToDo(parametersArr[0], false);
+    }
+
+    /**
+     * Add a Deadline to List of Task.
+     * @param description String name of deadline.
+     * @param by String represents DateTime the deadline need to be done by.
+     * @param isDone Status of the task (Done or Undone).
+     * @return Task that was added successfully.
+     * @throws ChatBotParameterException when String by is in wrong format.
+     */
+    public Task addDeadline(String description, String by, boolean isDone) throws ChatBotParameterException {
+        LocalDateTime byDateTime = Parser.parseDateTime(by);
+        Deadline deadline = new Deadline(description, byDateTime, isDone);
+        this.addTask(deadline);
+        return deadline;
+    }
+
+    /**
+     * Add a Deadline to List of Task.
+     * @param parameters String that contain all necessary fields to create a Deadline for adding.
+     * @return Task that was added successfully.
+     * @throws ChatBotParameterException when parameters String lack information to create a Deadline.
+     */
+    public Task addDeadline(String parameters) throws ChatBotParameterException {
+        String[] parametersArr = Parser.parseDeadline(parameters);
+        return this.addDeadline(parametersArr[0], parametersArr[1], false);
+
+    }
+
+    /**
+     * Add an Event to List of Task.
+     * @param parameters String that contain all necessary fields to create a Event for adding.
+     * @return Task that was added successfully.
+     * @throws ChatBotParameterException when parameters String lack information to create an Event.
+     */
+    public Task addEvent(String parameters) throws ChatBotParameterException {
+        String[] parametersArr = Parser.parseEvent(parameters);
+        return this.addEvent(parametersArr[0], parametersArr[1], parametersArr[2], false);
+    }
```

Comment:
I liked that your methods are clear and easy to understand!

Reply:
Thanks!